components/script/dom/permissions.rs

```diff
@@ -269,16 +268,22 @@ impl PermissionAlgorithm for Permissions {
             // Step 3.
             PermissionState::Prompt => {
                 let perm_name = status.get_query();
-                // https://w3c.github.io/permissions/#request-permission-to-use (Step 3 - 4)
-                let state = prompt_user(&format!(
-                    "{} {} ?",
-                    REQUEST_DIALOG_MESSAGE,
-                    perm_name.clone()
-                ));
-
+                
                 let globalscope = GlobalScope::current().expect("No current global object");
-                globalscope
-                    .as_window()
+
+                let globalscope_window = globalscope.as_window();
```

Comment:
This will cause us to panic when using Permissions from non-Window globals like workers. We should add is_headless to GlobalScope instead, which will let us avoid this.

Reply:
Double checking here - I will also have to add these fields to `WorkerGlobalScopeInit`, right? Which, in turn, is initialized with `script_thread` and used when inheriting `GlobalScope` from `WorkerGlobalScope` and `WorkletGlobalScope`.

Reply:
Correct.